internal/graph/singleflight_resolver.go

```diff
@@ -78,6 +84,9 @@ func (s *singleflightCheckResolver) ResolveCheck(
 		resp.ResolutionMetadata.DatastoreQueryCount = 0
 	}
 
+	span.SetAttributes(attribute.Bool("is_shared_request", shared))
```

Comment:
Can we rename to `deduplicated`?

Reply:
`shared` != deduplicated, it also includes the unique request that was ultimately resolved. 

If we want to know if a request was deduplicated, we would check `shared && !unique`. There are three states to communicate:
- Not shared, resolved as normal
- Shared and ultimately resolved
- Shared but deduplicated

Instead of communicating this as two booleans, how do you feel about combining into a single field?

Reply:
🤔 I guess `deduplicated` may imply the work encapsulated by this span was de-duplicated, but it actually wasn't unless `isUnique` is true.

So I think I'd lean toward setting an attr `deduplicated: false` if `isUnque==true` and setting `deduplicated: true` if `isUnique == false && shared`. Just capture the semantics of the de-dupe in a single attr.